cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -145,12 +151,35 @@
                                                      arguments.getOption(ProgramOptionConstants.RUN_ID));
     this.runId = RunIds.fromString(options.getArguments().getOption(ProgramOptionConstants.RUN_ID));
     this.metricsCollectionService = metricsCollectionService;
-    this.datasetFramework = datasetFramework;
+    this.datasetFramework
+      = createForWardingDatasetFramework(datasetFramework,
+                                         options.getArguments().getOption(ProgramOptionConstants.RUN_ID));
```

Comment:
It's better to use `runId.getId()`. Same applies to the creation of `loggingContext` above.